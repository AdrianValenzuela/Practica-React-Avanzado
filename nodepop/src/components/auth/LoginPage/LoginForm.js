// libraries imports
import React from 'react';
import PropTypes from 'prop-types';

// local imports
import { Button, FormField, Checkbox } from '../../shared';

function LoginForm({ onSubmit, isLoading }) {
 
    const [credentials, setCredentials] = React.useState({ email: '', password: '', remember: false });
    const handleFormFieldChange = event => {
        setCredentials(credentials => {
            return {
                ...credentials,
                [event.target.name]: event.target.type === 'remember' ? event.target.checked : event.target.value
            };
        });
    };

    const handleSubmitLoginForm = event => {
        event.preventDefault();
        onSubmit(credentials);
    };

    return (
        <form onSubmit={handleSubmitLoginForm}>
            <FormField 
                className={'input is-primary'} 
                type={'text'} 
                name={'email'}
                placeholder={'email'}
                value={credentials.email}
                onChange={handleFormFieldChange}
            />
            <FormField
                className={'input is-primary'} 
                type={'password'} 
                name={'password'}
                placeholder={'password'}
                value={credentials.password}
                onChange={handleFormFieldChange}
            />
            <Checkbox 
                className={'checkbox login-checkbox'}
                name={'remember'}
                type={'checkbox'}
                text={'Remember me'}
                disabled={!credentials.email || !credentials.password}
                checked={credentials.remember}
                onChange={handleFormFieldChange}
            />
            <Button 
                className={'button is-primary is-rounded'} 
                text={'Log in'}
                disabled={isLoading || !credentials.email || !credentials.password}
            />
        </form>
    );
}

LoginForm.propTypes = {
    onSubmit: PropTypes.func.isRequired,
    isLoading: PropTypes.bool
};

LoginForm.defaultProps = {
    isLoading: false
};

export default LoginForm;