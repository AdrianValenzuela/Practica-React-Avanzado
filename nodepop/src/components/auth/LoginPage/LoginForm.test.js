import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';

describe('LoginForm', () => {
    const props = {
        isLoading: false,
        onSubmit: jest.fn(),
    };


    const render = () => shallow(<LoginForm {...props} />);
    
    test('snapshot testing', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });
});