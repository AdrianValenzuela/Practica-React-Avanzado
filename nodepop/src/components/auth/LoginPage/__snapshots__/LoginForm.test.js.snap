// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot testing 1`] = `
<form
  onSubmit={[Function]}
>
  <FormField
    className="input is-primary"
    name="email"
    onChange={[Function]}
    placeholder="email"
    type="text"
    value=""
  />
  <FormField
    className="input is-primary"
    name="password"
    onChange={[Function]}
    placeholder="password"
    type="password"
    value=""
  />
  <Checkbox
    checked={false}
    className="checkbox login-checkbox"
    disabled={true}
    name="remember"
    onChange={[Function]}
    text="Remember me"
    type="checkbox"
  />
  <Button
    className="button is-primary is-rounded"
    disabled={true}
    text="Log in"
  />
</form>
`;
